nova_vm/src/ecmascript/types/language/object/property_key.rs

```diff
@@ -122,9 +122,26 @@ impl TryFrom<Value> for PropertyKey {
     fn try_from(value: Value) -> Result<Self, Self::Error> {
         match value {
             Value::Integer(x) => Ok(PropertyKey::Integer(x)),
+            Value::Float(x)
+                if x.fract() == 0.0
+                    && (SmallInteger::MIN_NUMBER..=SmallInteger::MAX_NUMBER)
```

Comment:
**nitpick:** Alright so we do need a special case for -0 apparently but it any other Float value being an Integer here should be an assert basically: As mentioned in the above discussion, the Venn diagram of the different number variants should not have any overlap: A safe integer should only ever be represented as a SmallInteger, never a Float with the exception of -0.